app/controllers/apps_controller.rb

```diff
@@ -90,60 +90,60 @@ def regenerate_api_key
-        email_at_notices = val.gsub(/-\d+/,"").split(",").map{|v| v.strip.to_i }.reject{|v| v == 0}
-        if email_at_notices.any?
-          params[:app][:email_at_notices] = email_at_notices
-        else
-          default_array = params[:app][:email_at_notices] = Errbit::Config.email_at_notices
-          flash[:error] = "Couldn't parse your notification frequency. Value was reset to default (#{default_array.join(', ')})."
-        end
+  # email_at_notices is edited as a string, and stored as an array.
+  def parse_email_at_notices_or_set_default
+    if params[:app] && (val = params[:app][:email_at_notices])
+      # Sanitize negative values, split on comma,
+      # strip, parse as integer, remove all '0's.
+      # If empty, set as default and show an error message.
+      email_at_notices = val.gsub(/-\d+/, "").split(",").map{|v| v.strip.to_i }.reject{|v| v == 0}
+      if email_at_notices.any?
+        params[:app][:email_at_notices] = email_at_notices
+      else
+        default_array = params[:app][:email_at_notices] = Errbit::Config.email_at_notices
+        flash[:error] = "Couldn't parse your notification frequency. Value was reset to default (#{default_array.join(', ')})."
       end
     end
+  end
 
-    def parse_notice_at_notices_or_set_default
-      if params[:app][:notification_service_attributes] && (val = params[:app][:notification_service_attributes][:notify_at_notices])
-        # Sanitize negative values, split on comma,
-        # strip, parse as integer, remove all '0's.
-        # If empty, set as default and show an error message.
-        notify_at_notices = val.gsub(/-\d+/,"").split(",").map{|v| v.strip.to_i }
-        if notify_at_notices.any?
-          params[:app][:notification_service_attributes][:notify_at_notices] = notify_at_notices
-        else
-          default_array = params[:app][:notification_service_attributes][:notify_at_notices] = Errbit::Config.notify_at_notices
-          flash[:error] = "Couldn't parse your notification frequency. Value was reset to default (#{default_array.join(', ')})."
-        end
+  def parse_notice_at_notices_or_set_default
+    if params[:app][:notification_service_attributes] && (val = params[:app][:notification_service_attributes][:notify_at_notices])
+      # Sanitize negative values, split on comma,
+      # strip, parse as integer, remove all '0's.
+      # If empty, set as default and show an error message.
+      notify_at_notices = val.gsub(/-\d+/, "").split(",").map{|v| v.strip.to_i }
+      if notify_at_notices.any?
+        params[:app][:notification_service_attributes][:notify_at_notices] = notify_at_notices
+      else
+        default_array = params[:app][:notification_service_attributes][:notify_at_notices] = Errbit::Config.notify_at_notices
+        flash[:error] = "Couldn't parse your notification frequency. Value was reset to default (#{default_array.join(', ')})."
       end
     end
+  end
 
-  private
+private
```

Comment:
Interestingly when you put it on one line the way I have, it does actually affect only that method. It works because in Ruby 2.1+, method definitions return their own name as a symbol and at that point private is just a method taking an argument. Since 'def app_params...' returns :app_params, the expression becomes `private :app_params` after the method definition is evaluated.